etc/scripts/smoketest.sh

```diff
@@ -0,0 +1,250 @@
+#
+
+# Smoke test a Helidon release. This assumes:
+# 1. The release has a source tag in the Helidon GitHub repo
+# 2. The bits are in either the OSS Sonatype Staging Repo or Maven Central
+# 3. You have a profile defined as "ossrh-staging" that configures
+#    https://oss.sonatype.org/content/groups/staging/ as a repository
+#    See bottom of RELEASE.md for details
+
+
+set -o pipefail || true  # trace ERR through pipes
+set -o errtrace || true # trace ERR through commands and functions
+set -o errexit || true  # exit the script if any statement returns a non-true return value
+
+on_error(){
+    CODE="${?}" && \
+    set +x && \
+    printf "[ERROR] Error(code=%s) occurred at %s:%s command: %s\n" \
+        "${CODE}" "${BASH_SOURCE}" "${LINENO}" "${BASH_COMMAND}"
+    echo "===== Log file: ${OUTPUTFILE} ====="
+    # In case there is a process left running
+}
+trap on_error ERR
+
+usage(){
+  cat <<EOF
+
+DESCRIPTION: Helidon Smoke Test Script
+
+USAGE:
+
+$(basename ${0}) [ --staged ] --version=V -workspace=path CMD
+
+  --staged
+        Use the OSS Sonatype Staging repository at
+        https://oss.sonatype.org/content/groups/staging/
+        If not specified assumes release is in Maven Central
+
+  --version=x.y.z
+        Test the specified version of Helidon
+
+  --clean
+        Remove the specified version of Helidon from the local
+        Maven repository cache before running tests. This is to
+        ensure artifacts are downloaded from remote repository.
+
+  --help
+        Prints the usage and exits.
+
+  CMD:
+
+    full
+        Runs full smoke test against released version:
+        1. archetypes
+        2. functional tests in workspace 
+        3. builds examples
+
+    quick
+        1. archetypes
+
+
+  Example:
+    smoketest.sh --staged --clean --version=1.4.1 full
+EOF
+}
+
+readonly SCRATCH=$(mktemp -d /var/tmp/helidon-smoke.XXXX) 
+
+# parse command line args
+ARGS=( "${@}" )
+for ((i=0;i<${#ARGS[@]};i++))
+{
+  ARG=${ARGS[${i}]}
+  case ${ARG} in
+  "--staged")
+    readonly STAGED_PROFILE="-Possrh-staging"
+    ;;
+  "--clean")
+    readonly CLEAN_MVN_REPO="true"
+    ;;
+  "--version="*)
+    VERSION=${ARG#*=}
+    ;;
+  "--help")
+    usage
+    exit 0
+    ;;
+  *)
+    if [ "${ARG}" = "full" ] || [ "${ARG}" = "quick" ] ; then
+      readonly COMMAND="${ARG}"
+    else
+      echo "ERROR: unknown argument: ${ARG}"
+      exit 1
+    fi
+    ;;
+  esac
+}
+
+if [ -z "${COMMAND}" ] ; then
+  echo "ERROR: no command provided"
+  exit 1
+fi
+
+if [ -z "${VERSION}" ] ; then
+  echo "ERROR: no version provided. Please use --version option to specify a version"
+  exit 1
+fi
+
+# Path to this script
+if [ -h "${0}" ] ; then
+  readonly SCRIPT_PATH="$(readlink "${0}")"
+else
+  readonly SCRIPT_PATH="${0}"
+fi
+
+full(){
+  echo "===== Full Test ====="
+  cd ${SCRATCH}
+  quick
+  cd ${SCRATCH}
+
+  echo "===== Cloning Workspace ====="
+  git clone https://github.com/oracle/helidon.git
```

Comment:
Can you get this from the current workspace ? This way if running this from a fork or a different repository this script is still usable.

E.g. `origin_url=$(git remote get-url origin)`

Reply:
Updated script to have a command line option, and if that isn't provide to default to the script's workspace origin.